### DISCUSSION TITLE
App crashed using Firebase distribution - IOS

### DISCUSSION THREAD

**Discussion by golanitay1984** [codemagic-ci-cd/codemagic-docs#1402] [Q&A]
Hi,

I am new to Codemagic so be gentle :)

I successfully uploaded my app for Android and IOS platforms using Firebase app distribution.
The app works fine on Android devices and also on IOS devices when I am debugging it,  but is not opened on IOS devices when I download it from Firebase.

I also not getting any errors during the process in Codemagic.

Is it a known issue? 
I'll appreciate any help,

Thanks!

**Comment by himesh-cm** [comment]
Hi @golanitay1984, Can you please share the build id?

**Reply by golanitay1984** [reply]
Please: 63eb3c06ce761d23e964ae16

**Reply by himesh-cm** [reply]
Hi @golanitay1984, Yes, the build looks ok. Crash can happen due to multiple reasons.
Are you catching any errors in your code and possibly pushing them on tools like `sentry` or `firebase crashlytics`?
Can you check this standard troubleshooting guide [here](https://blog.codemagic.io/testfligh-crash/)

**Reply by golanitay1984** [reply]
Hi,

I am not getting any errors from firebase crashlytics, I also follow the troubleshooting guide, but it is still not working.
I also tried to clean my app to include only minimal widgets.

**Reply by himesh-cm** [reply]
Are you using any firebase related dependencies? For example, is there any specific login on the start screen of your app which may lead to a crash? Can you share the logic?

**Reply by himesh-cm** [reply]
Hi @golanitay1984, Oversight from my end, you need to build via `ad_hoc` type and not `development` for distribution via firebase.

**Reply by golanitay1984** [reply]
Hi @himesh-cm , I am using only Firebase crashlytics and Analytics. There is no login page, It is a new app, so currently, I only Have 1 screen including only a single Text widget.

Can you please better explain how can I build the app via ad_hoc?

Thanks!

**Reply by himesh-cm** [reply]
Hi @golanitay1984, creating a new `ad_hoc profile` can be done via the apple developer portal following the same process, but the type is `ad_hoc`. 
Please try it and let me know if hit any roadblocks.

**Reply by golanitay1984** [reply]
Hi,

I created an ad_hoc profile and linked it to the CodeMagic build.
Still not working.

Thanks

**Reply by golanitay1984** [reply]
@himesh-cm, The problem was solved after I migrated my project to Swift and Kotlin.
Now I have another issue, but I'll open a new ticket.
